ratelimit/registry.go

```diff
@@ -23,10 +24,11 @@ type Registry struct {
 	global   Settings
 	//routeSettings map[string]Settings
 	lookup map[Settings]*Ratelimit
+	swarm  *swarm.Swarm
 }
 
 // NewRegistry initializes a registry with the provided default settings.
-func NewRegistry(settings ...Settings) *Registry {
+func NewRegistry(sw *swarm.Swarm, settings ...Settings) *Registry {
```

Comment:
on one hand, we typically try to avoid breaking, on the other hand, i don't think someone uses the ratelimit package separately. So the question is whether it is fine to refactor a bit the exported interface of this package? Because then i would have other suggestions, too.

If not, then we can just create a new function called NewSwarmRegistry, and the NewRegistry would simply call it with nil for the sw param.

Reply:
fixed